distribution1/dirtrbution.py

```diff
@@ -0,0 +1,236 @@
+        
+        line_coordinates_feet = np.array([(-3, 28), (0, 28), (3, 28), (0.5, 24)])
+        ri = [self.conductors[phase_cond]["r"]  for i in range(3)]
+        ri.append(self.conductors[neut_cond]["r"])
+        
+        gmri = [self.conductors[phase_cond]["gmr"]  for i in range(3)]
+        gmri.append(self.conductors[neut_cond]["gmr"])
+        
+        d = distance.cdist(line_coordinates_feet, line_coordinates_feet, 'euclidean')
+        for i in range(len(d)):
+            d[i,i] = gmri[i]
+        
+        
+        self.z = np.zeros(d.shape, dtype='complex')
+        r, c = self.z.shape
+        for i in range(r):
+            for j in range(c):
+                if i != j:
+                    self.z[i,j] = 0.09560 + 1j * 0.12134* (log(1/ d[i,j]) + 7.93402)
+                else:
+                    self.z[i,j] = ri[i] + 0.09560 + 1j * 0.12134* (log(1/ d[i,j]) + 7.93402)
+        
+        self.z = self.z * length_mi
+        self.kron_reduce()
+           
+        return
+    
+    def kron_reduce(self):
+        zij = self.z[:-1, :-1]
+        zin = self.z[:-1, -1]
+        zni = self.z[-1, :-1]
+        znn = self.z[-1, -1]
+        
+        Znn_inv = 1 / znn
+        self.zabc = zij - zin * Znn_inv * zni
+
+class Load:  
+    
+    V0 = 12470
+    
+    def __init__(self, P_base_zip, Q_base_zip, a0, a1, a2, b0, b1, b2):
+        self.P_base_zip = P_base_zip
+        self.Q_base_zip = Q_base_zip
+        self.a0= a0
+        self.a1 = a1
+        self.a2 = a2
+        self.b0 = b0
+        self.b1 = b1
+        self.b2 = b2
+
+        
+    def power(self, v:float):
+        P = self.P_base_zip * (self.a0  + self.a1 * (v/self.V0) + self.a2* (v/self.V0)**2)
+        Q = self.Q_base_zip * (self.b0  + self.b1 * (v/self.V0) + self.b2* (v/self.V0)**2)
+        return P + 1j* Q
+
+def destroy_federate(fed):
+    """
+    As part of ending a HELICS co-simulation it is good housekeeping to
+    formally destroy a federate. Doing so informs the rest of the
+    federation that it is no longer a part of the co-simulation and they
+    should proceed without it (if applicable). Generally this is done
+    when the co-simulation is complete and all federates end execution
+    at more or less the same wall-clock time.
+    :param fed: Federate to be destroyed
+    :return: (none)
+    """
+     # Adding extra time request to clear out any pending messages to avoid
+    #   annoying errors in the broker log. Any message are tacitly disregarded.
+    grantedtime = h.helicsFederateRequestTime(fed, h.HELICS_TIME_MAXTIME)
+    status = h.helicsFederateDisconnect(fed)
+    h.helicsFederateFree(fed)
+    h.helicsCloseLibrary()
+    logger.info("Federate finalized")
+
+def collect_data(data_list, idx, time, value):
+    data_list[idx][0].append(time)
+    data_list[idx][1].append(value)
+    return data_list
+
+
+if __name__ == "__main__":
+    
+    fed = h.helicsCreateValueFederateFromConfig("transmission_config.json")
```

Comment:
Should this be loading the `distribution_config.json` file instead?

```suggestion
    fed = h.helicsCreateValueFederateFromConfig("distribution_config.json")
```

Reply:
fixed